firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteIndexManager.java

```diff
@@ -233,6 +233,19 @@ public void deleteFieldIndex(FieldIndex index) {
     }
   }
 
+  @Override
+  public void createTargetIndices(Target target) {
```

Comment:
The problem for Or queries is they can only be treated as the whole. So either all `subTaregt` use indexes. Or neither of them.

For example, in `A || B` , only A has a full matched index. The SDK will run both `A` and `B` using full collection scan. After that, if the SDK finds it is more efficient to use indexes, it will try to create indexes for both `A` and `B`. So the next time this or query will be served using indexes.